superset/assets/src/dashboard/components/gridComponents/Tabs.jsx

```diff
@@ -204,45 +205,50 @@ class Tabs extends React.PureComponent {
               mountOnEnter
               unmountOnExit={false}
             >
-              {tabIds.map((tabId, tabIndex) => (
-                // react-bootstrap doesn't render a Tab if we move this to its own Tab.jsx so we
-                // use `renderType` to indicate what the DashboardComponent should render. This
-                // prevents us from passing the entire dashboard component lookup to render Tabs.jsx
-                <BootstrapTab
-                  key={tabId}
-                  eventKey={tabIndex}
-                  title={
-                    <DashboardComponent
-                      id={tabId}
-                      parentId={tabsComponent.id}
-                      depth={depth}
-                      index={tabIndex}
-                      renderType={RENDER_TAB}
-                      availableColumnCount={availableColumnCount}
-                      columnWidth={columnWidth}
-                      onDropOnTab={this.handleDropOnTab}
-                      onDeleteTab={this.handleDeleteTab}
-                    />
-                  }
-                >
-                  {renderTabContent && (
-                    <DashboardComponent
-                      id={tabId}
-                      parentId={tabsComponent.id}
-                      depth={depth} // see isValidChild.js for why tabs don't increment child depth
-                      index={tabIndex}
-                      renderType={RENDER_TAB_CONTENT}
-                      availableColumnCount={availableColumnCount}
-                      columnWidth={columnWidth}
-                      onResizeStart={onResizeStart}
-                      onResize={onResize}
-                      onResizeStop={onResizeStop}
-                      onDropOnTab={this.handleDropOnTab}
-                      isComponentVisible={selectedTabIndex === tabIndex}
-                    />
-                  )}
-                </BootstrapTab>
-              ))}
+              {tabIds.map((tabId, tabIndex) => {
+                const isTabContentVisible =
+                  selectedTabIndex === tabIndex &&
+                  (renderTabContent ? isComponentVisible : true);
```

Comment:
this simplifies to `!renderTabContent || isComponentVisible` I think. What does `renderTabContent` actually signify here? I'm having a bit of trouble understanding the logic.

Reply:
I updated the logic be more clear.